analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_dryspells.md

```diff
@@ -0,0 +1,827 @@
+```
+
+```python
+#plot the distribution of precipitation. Can clearly see that for leadtime=1 the values are more spread (but leadtime=1 is really the month that is currently occurring so less useful)
+g = sns.FacetGrid(df_for_quant, height=5, col="leadtime",col_wrap=3)
+g.map_dataframe(sns.histplot, "mean_cell",common_norm=False,kde=True,alpha=1,binwidth=10)
+
+for ax in g.axes.flatten():
+    ax.tick_params(labelbottom=True)
+    ax.set_ylabel("Number of months")
+    ax.set_xlabel("Total monthly precipitation (mm)")
+```
+
+```python
+#include all dates present in the observed rainfall df but not in the dry spell list, i.e. where no dryspells were observed, by merging outer
+df_ds_for=df_ds.merge(df_for_quant,how="right",on=["ADM1_EN","date_month"])
+df_ds_for.loc[:,"dry_spell"]=df_ds_for.dry_spell.replace(np.nan,0).astype(int)
+```
+
+```python
+df_ds_for["month"]=df_ds_for.date_month.dt.month
+df_ds_for["month_name"]=df_ds_for.month.apply(lambda x: calendar.month_name[x])
+df_ds_for["month_abbr"]=df_ds_for.month.apply(lambda x: calendar.month_abbr[x])
+df_ds_for_labels=df_ds_for.replace({"dry_spell":{0:"no",1:"yes"}}).sort_values("dry_spell",ascending=True)
+```
+
+```python
+#for some reason facetgrid doesn't want to show the values if there is only one occurence (i.e. in January..)
+g = sns.FacetGrid(df_ds_for_labels, height=5, col="leadtime",row="month_name",hue="dry_spell",palette={"no":"#CCE5F9","yes":'#F2645A'})
+g.map_dataframe(sns.histplot, "mean_cell",common_norm=False,alpha=1,binwidth=10)
+
+g.add_legend(title="Dry spell occurred")  
+for ax in g.axes.flatten():
+    ax.tick_params(labelbottom=True)
+    ax.set_ylabel("Number of months")
+    ax.set_xlabel("Total monthly precipitation (mm)")
+```
+
+```python
+#plot distribution precipitation with and withoud dry spell
+fig,ax=plt.subplots(figsize=(10,10))
+g=sns.boxplot(data=df_ds_for_labels,x="leadtime",y="mean_cell",ax=ax,color="#66B0EC",hue="dry_spell",palette={"no":"#CCE5F9","yes":'#F2645A'})
+ax.set_ylabel("Monthly precipitation")
+ax.spines['right'].set_visible(False)
+ax.spines['top'].set_visible(False)
+ax.set_xlabel("Leadtime")
+ax.get_legend().set_title("Dry spell occurred")
+# fig.savefig(os.path.join(plots_seasonal_dir,f"mwi_boxplot_formonth_dsobs_perlt_perc_{int(probability*100)}_{adm_str}_{month_str}.png"))
+```
+
+```python
+#compute tp,tn,fp,fn per threshold
+y_target =  df_ds_for.dry_spell
+threshold_list=np.arange(0,df_ds_for.mean_cell.max() - df_ds_for_labels.mean_cell.max()%10,10)
+df_pr_th=pd.DataFrame(threshold_list,columns=["threshold"]).set_index('threshold')
+for t in threshold_list:
+    y_predicted = np.where(df_ds_for.mean_cell<=t,1,0)
+
+    cm = confusion_matrix(y_target=y_target, 
+                          y_predicted=y_predicted)
+    tn,fp,fn,tp=cm.flatten()
+    df_pr_th.loc[t,["month_ds","month_no_ds","month_miss_rate","month_false_alarm_rate"]]=tp/(tp+fn+0.00001)*100,tn/(tn+fp+0.00001)*100,fn/(tp+fn+0.00001)*100,fp/(tp+fp+0.00001)*100
```

Comment:
It would help the readability if you kept the lines a bit shorter, to avoid needing to scroll horizontally

Reply:
Here I would define 0.0001 as a separate variable